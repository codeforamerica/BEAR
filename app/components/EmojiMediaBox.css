.emojiMediaContent {
  padding-left: 10px;
}
