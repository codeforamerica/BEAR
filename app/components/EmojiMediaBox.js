// @flow
import React, { Component } from 'react';
import styles from './EmojiMediaBox.css';

type Props = {
  className: string,
  emojiClass: string,
  title: string,
  content: string
};

export default class EmojiMediaBox extends Component<Props> {
  render() {
    const { className, emojiClass, title, content } = this.props;
    return (
      <div className={`${className} media-box`}>
        <div className="media-box__media media--icon">
          <div className={`emoji emoji--med ${emojiClass}`} />
        </div>
        <div className={`${styles.emojiMediaContent} media-box__content`}>
          <p>
            <b>{title}: </b>
            {content}
          </p>
        </div>
      </div>
    );
  }
}
