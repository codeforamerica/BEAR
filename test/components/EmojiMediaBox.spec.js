import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import EmojiMediaBox from '../../app/components/EmojiMediaBox';

Enzyme.configure({ adapter: new Adapter() });

describe('EmojiMediaBox component', () => {
  it('adds the emojiClass prop to the emoji div', () => {
    const component = shallow(<EmojiMediaBox emojiClass="emoji--baby" />);

    expect(component.find('.emoji').hasClass('emoji--baby')).toBe(true);
  });

  it('passes className to the rendered component', () => {
    const component = shallow(<EmojiMediaBox className="awesome-class" />);

    expect(component.hasClass('awesome-class')).toBe(true);
  });

  it('populates the title from props', () => {
    const component = shallow(<EmojiMediaBox title="I am a Title" />);

    expect(
      component
        .find('.media-box__content')
        .find('b')
        .text()
    ).toContain('I am a Title');
  });

  it('populates the content from props', () => {
    const component = shallow(<EmojiMediaBox content="Some content" />);

    expect(component.find('.media-box__content').text()).toContain(
      'Some content'
    );
  });

  it('should match exact snapshot', () => {
    const component = shallow(
      <EmojiMediaBox
        title="I am a Title"
        className="awesome-class"
        emojiClass="emoji--baby"
        content="Some content"
      />
    );
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
