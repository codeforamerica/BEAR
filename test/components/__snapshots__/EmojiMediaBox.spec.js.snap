// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmojiMediaBox component should match exact snapshot 1`] = `
<div
  className="awesome-class media-box"
>
  <div
    className="media-box__media media--icon"
  >
    <div
      className="emoji emoji--med emoji--baby"
    />
  </div>
  <div
    className="emojiMediaContent media-box__content"
  >
    <p>
      <b>
        I am a Title
        : 
      </b>
      Some content
    </p>
  </div>
</div>
`;
